serde/src/de/mod.rs

```diff
@@ -245,211 +307,147 @@ pub trait Deserializer {
     /// This method hints that the `Deserialize` type is expecting a `bool` value.
     #[inline]
     fn deserialize_bool<V>(&mut self, visitor: V) -> Result<V::Value, Self::Error>
-        where V: Visitor,
-    {
-        self.deserialize(visitor)
-    }
+        where V: Visitor;
 
     /// This method hints that the `Deserialize` type is expecting an `usize` value.
+    /// A reasonable default is to forward to `deserialize_u64` and cast the yielded value.
```

Comment:
I don't think "and cast the yielded value" accurately captures what is going on here. The Visitor has an associated type `Value: Deserialize` and it has a method `visit_u64` which gives back `Value`. If `deserialize_usize` receives a visitor whose `Value` is `usize`, and `deserialize_usize` forwards to `deserialize_u64`, and `deserialize_u64` calls `visit_u64` on the Visitor, really it would be the `Visitor` that is responsible for any casting to go from u64 to usize.

I would stick with "A reasonable default is to forward to \`deserialize_u64\`." Similar in the other methods.